R/quasiquotation.R

```diff
@@ -1,61 +1,80 @@
-#' Quasiquotation of an expression
+#' Forcing parts of an expression
 #'
 #' @description
 #'
-#' Quasiquotation is the mechanism that makes it possible to program
-#' flexibly with tidy evaluation grammars like dplyr. It is enabled in
-#' all functions quoting their arguments with `enquo()`, `enexpr()`,
-#' or the plural variants.
+#' It is sometimes useful to force early evaluation of part of an
+#' expression before it gets fully evaluated. The tidy eval framework
+#' provides several forcing operators for different use cases.
 #'
-#' Quasiquotation is the combination of quoting an expression while
-#' allowing immediate evaluation (unquoting) of part of that
-#' expression.
+#' - The bang-bang operator `!!` forces a _single_ object. One
+#'   common case for `!!` is to substitute an environment-variable
+#'   (created with `<-`) with a data-variable (inside a data frame).
 #'
-#' - The `!!` operator unquotes its argument. It gets evaluated
-#'   immediately in the surrounding context.
+#'   ```
+#'   library(dplyr)
 #'
-#' - The `!!!` operator unquotes and splices its argument. The
-#'   argument should represent a list or a vector. Each element will
-#'   be embedded in the surrounding call, i.e. each element is
-#'   inserted as an argument. If the vector is named, the names are
-#'   used as argument names.
+#'   # The environment variable `var` refers to the data-variable
+#'   # `height`
+#'   var <- sym("height")
 #'
-#'   If the vector is a classed object (like a factor), it is
-#'   converted to a list with [base::as.list()] to ensure proper
-#'   dispatch. If it is an S4 objects, it is converted to a list with
-#'   [methods::as()].
+#'   # We force `var`, which substitutes it with `height`
+#'   starwars %>%
+#'     summarise(avg = mean(!!var, na.rm = TRUE))
+#'   ```
 #'
-#' - The `{{ }}` operator quotes and unquotes its argument in one
-#'   go, a pattern that we call **interpolation**. It is an alias for
-#'   `!!enquo(arg)`.
+#' - The big-bang operator `!!!` forces-splice a _list_ of objects.
+#'   The elements of the list are spliced in place, meaning that they
+#'   each become one single argument.
 #'
-#'   Like `enquo()`, `{{ }}` is used in functions to capture an
-#'   argument as a quoted expression. This expression is immediately
-#'   unquoted in place.
+#'   ```
+#'   vars <- syms(c("height", "mass"))
+#'
+#'   # Force-splicing is equivalent to supplying the elements separately
+#'   starwars %>% select(!!!vars)
+#'   starwars %>% select(height, mass)
+#'   ```
+#'
+#' - The curly-curly operator `{{ }}` for function arguments is a bit
```

Comment:
I'd suggest something like: this is useful when you have a data-variable in a env-variable (i.e. a function argument)